src/flamenco/runtime/program/fd_native_cpi.c

```diff
@@ -57,8 +57,7 @@ fd_native_cpi_execute_system_program_instruction( fd_exec_instr_ctx_t * ctx,
         if( acct_meta->is_writable ) {
           instr_info->acct_flags[j] |= FD_INSTR_ACCT_FLAGS_IS_WRITABLE;
         }
-        // TODO: should check the parent has signer flag set
-        if( acct_meta->is_signer ) {
+        if( acct_meta->is_signer && fd_instr_acc_is_signer( ctx->instr, (const fd_pubkey_t *)acct_meta->pubkey) ) {
```

Comment:
I actually think the better way to handle this would be to iterate through all of the accounts for the instruction and use the fd_instr_is_signer_idx() method